### PR TITLE
Metadata support

### PR DESCRIPTION
**Pull request by vdancik**
This serves as replacement for /predicates endpoint. Addresses issue #143 and replaces PR #171.

### CODE REVIEW

File: TranslatorReasonerAPI.yaml [hunk 1]
```diff
@@ -55,6 +56,19 @@ paths:
                     'biolink:Gene':
                       - biolink:directly_interacts_with
                       - biolink:decreases_activity_of
+  /knowledge_map:
```
Comment (patrickkwang) [inline TranslatorReasonerAPI.yaml:59]: Re: naming, I'd prefer meta_graph or meta_knowledge_graph. That would align with the SmartAPI/BTE notion of a MetaKG. It's not obvious to me that the phrase "knowledge map" would be interpreted by a naive reader much differently from "knowledge graph".

File: TranslatorReasonerAPI.yaml [hunk 2]
```diff
@@ -579,3 +593,97 @@ components:
...
+      properties:
+        id_prefixes:
+          type: array
+          description: >-
+            List of CURIE prefixes for the node category that this TRAPI web
+            service understands and accepts on the input.
+          items:
+            type: string
+          example: ['ChEMBL:', 'inchikey:']
+        count:
```
Comment (patrickkwang) [inline TranslatorReasonerAPI.yaml:627]: I'm not sure that/how the Ranking Agent would use count information if it were provided, so I'm inclined to put it in the extended schema.

File: TranslatorReasonerAPI.yaml [hunk 3]
```diff
@@ -579,3 +593,97 @@ components:
...
+        Description of a node category provided by this TRAPI web service.
+      properties:
+        id_prefixes:
+          type: array
+          description: >-
+            List of CURIE prefixes for the node category that this TRAPI web
+            service understands and accepts on the input.
+          items:
+            type: string
+          example: ['ChEMBL:', 'inchikey:']
```
Comment (patrickkwang) [inline TranslatorReasonerAPI.yaml:626]: We should align all of the examples with the biolink context prefixes: https://github.com/biolink/biolink-model/blob/master/context.jsonld

In this case, we may want something like `['CHEMBL.COMPOUND', 'INCHIKEY']`.

Also, should these prefixes include the trailing colon? My feeling is no.

Comment (edeutsch) [reply TranslatorReasonerAPI.yaml:626]: I agree, examples should show official curie prefixes without colons

File: TranslatorReasonerAPI.yaml (hunk 1 above)
Comment (edeutsch) [reply TranslatorReasonerAPI.yaml:59]: The results on the last vote on this are in the minutes from Dec 10. They were quite split although /knowledge_map was the winner with a slight majority. I would be fine with reconsidering and revoting on the endpoint name as we complete this for real.

File: TranslatorReasonerAPI.yaml (hunk 3 above)
Comment (kevinxin90) [reply TranslatorReasonerAPI.yaml:626]: Wondering for id_prefixes, does the order an id prefix appears in the list indicate priority?

Comment (vdancik) [reply TranslatorReasonerAPI.yaml]: There is no intent to indicate preference, service promises to handle any of the prefixes.

File: TranslatorReasonerAPI.yaml (hunk 2 above)
Comment (vdancik) [reply TranslatorReasonerAPI.yaml]: done

File: TranslatorReasonerAPI.yaml [hunk 4]
```diff
@@ -579,3 +594,70 @@ components:
...
+      type: object
+      description: >-
+        Description of a node category provided by this TRAPI web service.
+      properties:
+        id_prefixes:
+          type: array
+          description: >-
+            List of CURIE prefixes for the node category that this TRAPI web
+            service understands and accepts on the input.
+          items:
```
Comment (patrickkwang) [inline TranslatorReasonerAPI.yaml:625]: Should we insist that this array be non-empty?

File: TranslatorReasonerAPI.yaml [hunk 5]
```diff
@@ -579,3 +594,70 @@ components:
...
+          additionalProperties:
+            $ref: '#/components/schemas/KMNode'
+        edges:
+          type: array
+          description: >-
+            List of edges/predicates provided by this TRAPI web service.
+          items:
+            $ref: '#/components/schemas/KMEdge'
+    KMNode:
+      type: object
```
Comment (patrickkwang) [inline TranslatorReasonerAPI.yaml:616]: These values are objects with exactly one property, which begs the question, should the values just be the id-prefix arrays, directly?

Comment (vdancik) [reply TranslatorReasonerAPI.yaml:616]: I would say no, we will also have attributes here as well

File: TranslatorReasonerAPI.yaml (hunk 4 above)
Comment (vdancik) [reply TranslatorReasonerAPI.yaml:625]: Makes sense, actually the array of edges in the meta knowledge graph should be non-empty as well.

File: TranslatorReasonerAPI.yaml [hunk 6]
```diff
@@ -579,3 +594,70 @@ components:
...
+      description: >-
+        Description of a node category provided by this TRAPI web service.
+      properties:
+        id_prefixes:
+          type: array
+          description: >-
+            List of CURIE prefixes for the node category that this TRAPI web
+            service understands and accepts on the input.
+          items:
+            type: string
```
Comment (patrickkwang) [inline TranslatorReasonerAPI.yaml:626]:
```suggestion
            type: string
          minItems: 1
```

File: TranslatorReasonerAPI.yaml [hunk 7]
```diff
@@ -579,3 +594,70 @@ components:
...
+            Collection of node categories provided by this TRAPI web service,
+            indexed by Biolink class CURIEs.
+          additionalProperties:
+            $ref: '#/components/schemas/KMNode'
+        edges:
+          type: array
+          description: >-
+            List of edges/predicates provided by this TRAPI web service.
+          items:
+            $ref: '#/components/schemas/KMEdge'
```
Comment (patrickkwang) [inline TranslatorReasonerAPI.yaml:614]:
```suggestion
            $ref: '#/components/schemas/KMEdge'
          minItems: 1
```

File: TranslatorReasonerAPI.yaml [hunk 8]
```diff
@@ -579,3 +594,72 @@ components:
...
+          minItems: 1
+    KMNode:
```
Comment (patrickkwang) [inline TranslatorReasonerAPI.yaml:616]: We should probably rename these to MKNode and MKEdge, or just MNode and MEdge.

Comment (vdancik) [reply TranslatorReasonerAPI.yaml:616]: Or maybe MetaNode, MetaEdge to complement MetaKnowledgeGraph

**Review by patrickkwang — APPROVED** [review]

**Review by edeutsch — APPROVED** [review]

**Review by CaseyTa — APPROVED** [review]

**Review by rtroper — APPROVED** [review]

**Comment by sstemann** [comment]
@NCATSTranslator/trapi-wg added reviewers, per the 3/11 TRAPI WG call, the WG is looking for review/approval/dissent

**Review by bill-baumgartner — APPROVED** [review]

**Review by marcdubybroad — APPROVED** [review]

**Review by brettasmi — APPROVED** [review]
👍

**Comment by Lilly-25** [comment]
Hello,
Can someone guide me on how to download the knowledge graph here?

**Pull request merged**